GDJS/GDJS/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -1096,33 +1076,35 @@ gd::String EventsCodeGenerator::GenerateConditionsListCode(
     gd::EventsCodeGenerationContext& context) {
   gd::String outputCode;
 
-  for (std::size_t i = 0; i < conditions.size(); ++i)
     outputCode += GenerateBooleanInitializationToFalse(
-        "condition" + gd::String::From(i) + "IsTrue", context);
+        "isConditionTrue", context);
 
   for (std::size_t cId = 0; cId < conditions.size(); ++cId) {
-    if (cId != 0)
-      outputCode +=
-          "if ( " +
-          GenerateBooleanFullName(
-              "condition" + gd::String::From(cId - 1) + "IsTrue", context) +
-          ".val ) {\n";
-
     gd::String conditionCode =
         GenerateConditionCode(conditions[cId],
-                              "condition" + gd::String::From(cId) + "IsTrue",
+                              "isConditionTrue",
                               context);
     if (!conditions[cId].GetType().empty()) {
+      for (std::size_t i = 0; i < cId;
+           ++i)  // Skip conditions if one condition is false. //TODO : Can be
+                 // optimized
+      {
+        if (i == 0)
+          outputCode += "if (";
+        else
+          outputCode += " && ";
+        outputCode += GenerateBooleanFullName("isConditionTrue", context);
+        if (i == cId - 1) outputCode += ") ";
+      }
+
       outputCode += "{\n";
+      outputCode +=
+          GenerateBooleanFullName("isConditionTrue", context) + " = false;\n";
       outputCode += conditionCode;
-      outputCode += "}";
+      outputCode += "}\n";
     }
   }
 
-  for (std::size_t cId = 0; cId < conditions.size(); ++cId) {
-    if (cId != 0) outputCode += "}\n";
```

Comment:
This might be a micro optimization, but this allowed to directly jump outside of conditions when one was false.

Reply:
I'm not sure to understand, is it to avoid to run N if when the 1st condition is false?
The only way I see is to nest the if but it will look odd.

Reply:
Yes this is to avoid N `if`s when the 1st condition is false.
This might be a micro-optimisation because the resulting machine code is probably equivalent (multiple branches to jump through)... or not? Nesting if is not pretty but might help generating code that jumps directly at the end if a condition is false.

Probably not a good idea to hack this with a `while` and `break`. Sure it could work, but it will probably do the job of optimising the code harder for the JS engine.